# FIXES THIS WARNING (made by arthurx07)
#dmenu.c:740:39: warning: passing argument 1 of ‘free’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
#  740 |                         free(colors[j][i]);
#      |                              ~~~~~~~~~^~~
#In file included from dmenu.c:6:
#/usr/include/stdlib.h:687:25: note: expected ‘void *’ but argument is of type ‘const char *’
#  687 | extern void free (void *__ptr) __THROW;
#      |                   ~~~~~~^~~~~
#dmenu.c: In function ‘main’:
#dmenu.c:947:15: warning: passing argument 1 of ‘free’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
#  947 |     free(fonts[0]);
#      |          ~~~~~^~~
#/usr/include/stdlib.h:687:25: note: expected ‘void *’ but argument is of type ‘const char *’
#  687 | extern void free (void *__ptr) __THROW;
#      |                   ~~~~~~^~~~~
diff --git a/dmenu.c b/dmenu.c
index 26bd8c0..098e291 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -737,7 +737,7 @@ setup(void)
 	}
 	for (j = 0; j < SchemeOut; ++j) {
 		for (i = 0; i < 2; ++i)
-			free(colors[j][i]);
+			free((char*)colors[j][i]);
 	}
 
 	clip = XInternAtom(dpy, "CLIPBOARD",   False);
@@ -944,7 +944,7 @@ main(int argc, char *argv[])
 	if (!drw_fontset_create(drw, (const char**)fonts, LENGTH(fonts)))
 		die("no fonts could be loaded.");
 
-    free(fonts[0]);
+    free((char*)fonts[0]);
 	lrpad = drw->fonts->h;
 
 	sp = sidepad;
